erdjs/src/smartcontracts/interaction/interaction.ts

```diff
@@ -0,0 +1,198 @@
+import { Balance } from "../../balance";
+import { GasLimit } from "../../networkParams";
+import { Transaction } from "../../transaction";
+import { TransactionOnNetwork } from "../../transactionOnNetwork";
+import { Query } from "../query";
+import { QueryResponse } from "../queryResponse";
+import { ContractFunction } from "../function";
+import { Address } from "../../address";
+import { SmartContract } from "../smartContract";
+import { IInteractionRunner } from "./interface";
+import { EndpointDefinition, TypedValue } from "../typesystem";
+import { Nonce } from "../../nonce";
+import { ImmediateResult, SmartContractResults } from "../smartContractResults";
+import { ReturnCode } from "../returnCode";
+
+/**
+ * Interactions are mutable (the interaction runner mutates their content: transaction, query).
+ * But they aren't really reusable, therfore their mutability should not cause erroneous usage (generally speaking).
+ */
+
+export class Interaction {
```

Comment:
Would be helpful to have a short description of the relationships between a smart contract, its ABI, an interactor, an interaction and an interaction runner.

Reply:
Maybe in the PR description.

Reply:
A `SmartContract` is provided an `SmartContractAbi` - the ABI is comprised of one or more `ContractInterface` objects. A `SmartContractAbi` is loaded from an `AbiRegistry` (which, ultimately, is loaded from a JSON file).

A `SmartContract` provided with the ABI exposes its endpoints as JavaScript functions on the `contract.methods` object. These functions, when called, create and return the so-called `Interaction` objects. These objects act as builders for `Transaction` and `Query` objects.

An `InteractionRunner` is able to `run` (with or without waiting after the SCRs), `run as query` or `run as simulation` an `Interaction` object.